### PR TITLE
doc(lxl): add buffer api description

### PR DESCRIPTION
**Pull request by lixl-st**
## Check List

- [ ] merge the latest version source branch/repo, and resolve all the conflicts
- [ ] pass style check
- [ ] pass all the tests

### CODE REVIEW

File: ding/data/buffer/deque_buffer.py [hunk 1]
```diff
@@ -47,8 +47,18 @@ def clear(self):
...
     def __init__(self, size: int) -> None:
+        """
+        Overview:
+            The initialization method of DequeBuffer.
+        Arguments:
+            - size (:obj:`int`): The maximum number of objects that the buffer can hold.
+        """
         super().__init__(size=size)
         self.storage = deque(maxlen=size)
         # Meta index is a dict which use deque as values
```
Comment (Will-Nie) [inline ding/data/buffer/deque_buffer.py:64]: a dict which uses

File: ding/data/buffer/deque_buffer.py [hunk 2]
```diff
@@ -57,6 +67,14 @@ def __init__(self, size: int) -> None:
 
     @apply_middleware("push")
     def push(self, data: Any, meta: Optional[dict] = None) -> BufferedData:
+        """
+        Overview:
+            The method that input the objects and related meta information into the buffer.
```
Comment (Will-Nie) [inline ding/data/buffer/deque_buffer.py:72]: ``the related meta``

File: ding/data/buffer/middleware/clone_object.py [hunk 3]
```diff
@@ -5,9 +5,10 @@
 
 def clone_object():
     """
-    This middleware freezes the objects saved in memory buffer as a copy,
-    try this middleware when you need to keep the object unchanged in buffer, and modify
-    the object after sampling it (usually in multiple threads)
+    Overview:
+        This middleware freezes the objects saved in memory buffer as a copy,
```
Comment (Will-Nie) [inline ding/data/buffer/middleware/clone_object.py:9]: full stop after copy

File: ding/data/buffer/middleware/sample_range_view.py [hunk 4]
```diff
@@ -5,6 +5,13 @@
 
 
 def sample_range_view(buffer_: 'Buffer', start: Optional[int] = None, end: Optional[int] = None) -> Callable:
+    """
+    Overview:
+        The middleware that places restrictions on indices range when sampling.
```
Comment (Will-Nie) [inline ding/data/buffer/middleware/sample_range_view.py:10]: ``indices' range``

File: ding/data/buffer/deque_buffer.py [hunk 5]
```diff
@@ -47,8 +47,18 @@ def clear(self):
```
Comment (Will-Nie) [inline ding/data/buffer/deque_buffer.py:47]: fix the unittest failure

File: ding/data/buffer/deque_buffer.py [hunk 6]
```diff
@@ -154,22 +210,46 @@ def delete(self, indices: Union[str, Iterable[str]]) -> None:
...
+        Overview:
+            The method that returns the current length of the buffer.
+        """
         return len(self.storage)
 
     def get(self, idx: int) -> BufferedData:
+        """
+        Overview:
+            The method that returns the object given a specific index.
+        """
```
Comment (zhangpaipai) [inline ding/data/buffer/deque_buffer.py:223]: maybe add Returns discription or specify the object(BufferedData) is better?

File: ding/data/buffer/middleware/priority.py [hunk 7]
```diff
@@ -18,6 +22,18 @@ def __init__(
...
             IS_weight_anneal_train_iter: int = int(1e5),
     ) -> None:
+        """
+        Arguments:
+            - buffer (:obj:`Buffer`): The buffer to use PER.
+            - IS_weight (:obj:`bool`): Whether use importance sampling or not.
+            - priority_power_factor (:obj:`float`): The factor that adjust the sensitivity between\
+                the sampling probability and the priority level.
+            - IS_weight_power_factor (:obj:`float`): The factor that adjust the sensitivity between\
+                the sample rarity and sampling probability in importance sampling.
```
Comment (zhangpaipai) [inline ding/data/buffer/middleware/priority.py:32]: maybe adding the greek symbols like beta in the article is better?

File: ding/data/buffer/middleware/priority.py [hunk 8]
```diff
@@ -18,6 +22,18 @@ def __init__(
...
+            - IS_weight_anneal_train_iter (:obj:`float`): The factor that controls the increasing of\
+                ``IS_weight_power_factor`` during training.
+        """
```
Comment (zhangpaipai) [inline ding/data/buffer/middleware/priority.py:35]: maybe explaining the self.pivot is better?

File: ding/data/buffer/middleware/priority.py [hunk 9]
```diff
@@ -18,6 +22,18 @@ def __init__(
             IS_weight_power_factor: float = 0.4,
             IS_weight_anneal_train_iter: int = int(1e5),
     ) -> None:
+        """
+        Arguments:
+            - buffer (:obj:`Buffer`): The buffer to use PER.
+            - IS_weight (:obj:`bool`): Whether use importance sampling or not.
+            - priority_power_factor (:obj:`float`): The factor that adjust the sensitivity between\
+                the sampling probability and the priority level.
```
Comment (zhangpaipai) [inline ding/data/buffer/middleware/priority.py:30]: is it the alpha in the article?

File: ding/data/buffer/middleware/priority.py (hunk 7 above)
Comment (lixl-st) [reply ding/data/buffer/middleware/priority.py:32]: Good suggestions! While the naming of those parameters needs to consider both simplicity and clarity, we may decide to rename it in the future.

**Pull request merged**